### PR TITLE
Made searching speed about 10x faster

### PR DESCRIPTION
**Pull request by Aztup**
Please dont mind the code it's a little bit dirty it's my second time dealing with extensions haha.

### CODE REVIEW

File: chrome/content.js [hunk 1]
```diff
@@ -12,15 +14,42 @@ const USER = {
   ERROR: getURL('images/user-error.png'),
 };
 
-const get = url => fetch(`https://${url}`).then(res => res.json());
+async function sleep(time) {
+  return new Promise((res) => {
+    setTimeout(res, time * 1000);
+  });
+}
```
Comment (matthewfripp) [inline chrome/content.js:21]:
```suggestion
const sleep = time => new Promise(res => setTimeout(res, time * 1000));
```

File: chrome/content.js [hunk 2]
```diff
@@ -12,15 +14,42 @@ const USER = {
...
-const post = (url, body) => fetch(`https://${url}`, {
-  method: 'POST',
-  headers: {
-    'Content-Type': 'application/json',
-  },
-  body,
-}).then(res => res.json());
+const get = (async (url) => {
+  try {
+    const request = await fetch(`https://${url}`)
```
Comment (matthewfripp) [inline chrome/content.js:25]:
```suggestion
    const request = await fetch(`https://${url}`);
```

File: chrome/content.js [hunk 3]
```diff
@@ -12,15 +14,42 @@ const USER = {
...
+}
 
-const post = (url, body) => fetch(`https://${url}`, {
-  method: 'POST',
-  headers: {
-    'Content-Type': 'application/json',
-  },
-  body,
-}).then(res => res.json());
+const get = (async (url) => {
```
Comment (matthewfripp) [inline chrome/content.js:23]:
```suggestion
const get = async url => {
```

File: chrome/content.js [hunk 4]
```diff
@@ -12,15 +14,42 @@ const USER = {
...
+  try {
+    const request = await fetch(`https://${url}`)
+    if (!request.ok) throw new Error('Request failed');
+
+    return await request.json();
+  } catch (error) {
+    await sleep(0.2);
+    return await get(url);
+  }
+});
```
Comment (matthewfripp) [inline chrome/content.js:33]:
```suggestion
};
```

File: chrome/content.js [hunk 5]
```diff
@@ -12,15 +14,42 @@ const USER = {
...
+
+const post = (async (url, body) => {
```
Comment (matthewfripp) [inline chrome/content.js:35]:
```suggestion
const post = async (url, body) => {
```

File: chrome/content.js [hunk 6]
```diff
@@ -12,15 +14,42 @@ const USER = {
...
+    });
+
+    if (!request.ok) throw new Error('Request failed');
+
+    return await request.json();
+  } catch (error) {
+    await sleep(0.2);
+    return await post(url, body);
+  }
+});
```
Comment (matthewfripp) [inline chrome/content.js:52]:
```suggestion
};
```

File: chrome/content.js [hunk 7]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable no-loop-func */
+/* eslint-disable no-return-await */
```
Comment (matthewfripp) [inline chrome/content.js:2]:
```suggestion
```
Add these exemptions to the eslint file instead please

File: chrome/content.js [hunk 8]
```diff
@@ -45,77 +74,147 @@ input.oninput = () => {
...
+    });
+  }
 
+  if (targetServerId) {
+    icon.src = getURL('images/user-success.png');
     color(COLORS.GREEN);
-    setTimeout(() => color(COLORS.BLUE), 1000);
+    setTimeout(() => {
+      color(COLORS.BLUE);
+    }, 1000);
```
Comment (matthewfripp) [inline chrome/content.js:162]:
```suggestion
    setTimeout(() => color(COLORS.BLUE), 1000);
```

File: chrome/content.js [hunk 9]
```diff
@@ -45,77 +74,147 @@ input.oninput = () => {
 
 let searching = false;
 let canceled = false;
+let foundAllServers = false;
+let searchingTarget = true;
+let allPlayers = [];
+let playersCount = 0;
+let targetsChecked = 0;
+let targetServerId = '';
+let maxPlayers = 0;
```
Comment (matthewfripp) [inline chrome/content.js:83]:
```suggestion
let foundAllServers = false;
let searchingTarget = true;
let playersCount = 0;
let targetsChecked = 0;
let targetServerId = '';
let maxPlayers = 0;
```

File: chrome/content.js [hunk 10]
```diff
@@ -45,77 +74,147 @@ input.oninput = () => {
...
 
-async function find(imageUrl, place, cursor = '', count = 0) {
-  status.innerText = 'Searching...';
-  color(COLORS.BLUE);
-  search.src = getURL('images/cancel.png');
-  icon.src = getURL('images/user-success.png');
-  input.disabled = true;
+const allThumbnails = new Map();
```
Comment (matthewfripp) [inline chrome/content.js:85]:
```suggestion
const allPlayers = [];
const allThumbnails = new Map();
```

**Review by matthewfripp — CHANGES_REQUESTED** [review]
Thanks so much for helping!! Just added a few suggestions

**Comment by Aztup** [comment]
Everything should be good now :)

**Comment by matthewfripp** [comment]
Yes! looks great. A way to speed it up even further would be to cache page cursors, they are static surprisingly

**Comment by Aztup** [comment]
Looks interesting but I'm pretty sure it doesnt take that long to get all servers. It could save up extra second tho

**Pull request merged**